runtime/.clang-tidy

```diff
@@ -0,0 +1,231 @@
+---
```

Comment:
Just curious, did we copy this from somewhere and/or where there adjustments?

Reply:
This is copied from [the PL-Lightning repo](https://github.com/PennyLaneAI/pennylane-lightning/blob/master/pennylane_lightning/src/.clang-tidy). We follow the same style.